pkg/client/didexchange/client.go

```diff
@@ -38,6 +36,10 @@ type Client struct {
 	inboundTransportEndpoint string
 	actionCh                 chan dispatcher.DIDCommAction
 	msgCh                    chan dispatcher.StateMsg
+	actionEvent              chan<- dispatcher.DIDCommAction
+	msgEvents                []chan<- dispatcher.StateMsg
+	lock                     sync.RWMutex
+	msgEventLock             sync.RWMutex
```

Comment:
better to keep the mutex close to the data it is destined to protect
`msgEventLock` field should be above `msgEvents` the point is we can see what this mutex for
e.g
```
lock                          sync.RWMutex	
actionEvent              chan<- dispatcher.DIDCommAction
msgEventLock         sync.RWMutex
msgEvents               []chan<- dispatcher.StateMsg
```
in this structure, we can see that `lock` is responsible for `actionEvent` and `msgEventLock` for `msgEvents`

Reply:
and probably better to rename `lock` to `actionEventLock`

Reply:
[Done](https://github.com/hyperledger/aries-framework-go/compare/7f2cd61c8bc3526f13b02f38f39e9946c1881f10..7e659b43cc8ebdfacb141bb9a0805dcddc74b450#diff-427522b353a28b1ec350c29c79d387b0R42).